lib/pmd.js

```diff
@@ -179,6 +203,34 @@ pmd.processFile = function(file){
       }//i in entity.types
     }//entity.code && entity.type === pmd.DOCTYPES.CONSTANTS
 
+    if (entity.code && entity.type === pmd.DOCTYPES.VARIABLES){
+      entity.variables = jsonData.ctx.variables;
+
+      // Loop over constants to see if there's one for this constantName
+      for (var i in entity.variables){
+        entity.variables[i].isPrivate = entity.isPrivate;
+        variables.forEach(function(varType){
+          if (entity.variables[i].name === varType.name){
+            entity.variables[i].description = varType.description
+          }
+        });//tagConstants.forEach
+      }//i in entity.types
+    }//entity.code && entity.type === pmd.DOCTYPES.CONSTANTS
+
+    if (entity.code && entity.type === pmd.DOCTYPES.EXCEPTIONS){
+      entity.exceptions = jsonData.ctx.exceptions;
+
+      // Loop over constants to see if there's one for this constantName
```

Comment:
Changed it over, maybe there is a way to use an Object Oriented way to process the variables, constants and exceptions because they all use the same logic. Only some things are different (regex and the array where to append it to).